src/Responses/Files/RetrieveResponse.php

```diff
@@ -28,7 +28,7 @@ private function __construct(
         public readonly string $filename,
         public readonly string $purpose,
         public readonly string $status,
-        public readonly ?array $statusDetails,
+        public readonly ?mixed $statusDetails,
```

Comment:
What's the PHP's type we can have here?

Reply:
Hi @nunomaduro , The data that I currently have as a basis for this is not yet clear, if the training files were uploaded successfully, I only got `null` for `statusDetails`, but the only time it has value is when there's an error and the data type is `string` for the exception message from OpenAI.

Should I change it to `public readonly ?string $statusDetails` instead? I'm asking you this since you far much more knowledgeable and I want to know what works best for you.

Reply:
@rodrigoisabelo I tried to figure out what the correct type should be, but the API specification is not very clear what we can expect here: https://github.com/openai/openai-openapi/blob/master/openapi.yaml#L3042

The spec says it is an `object` (without further clarification) or `null`. Therefore the current implementation with `?array` seems to be reasonable.

Furthermore I tried to reproduce your issue where it returns a string instead of null (or an object) but I was not able to do so as I always got `null`.

Can you provide us a way to reproduce the issue?

Reply:
Try replacing the extension of this file into JSONL and upload it on OpenAI files API. It might take some time to produce an error. But from what have experienced the status details were an exception message from them.

[invalid_training_data.txt](https://github.com/openai-php/client/files/10866294/invalid_training_data.txt)

Reply:
@rodrigoisabelo I am not able to upload this file to OpenAI. Already during the upload I've got an error response:

```
OpenAI\Exceptions\ErrorException {
  #message: "Expected file to have JSONL format, where every line is a JSON dictionary. Line 2 is not a dictionary (HINT: line starts with: "{"pr...")."
  #code: 0
  #file: "/Users/sandro/Code/openai-php-client/src/Transporters/HttpTransporter.php"
  #line: 61
  ...
```

Probably you run into some strange malfunctioning on the API side. As long it is not possible to reproduce, I don't think it's worth to fix it on client side especially the change would go against their API documentation.

Reply:
It's fine @gehrisandro , it's not that we are always uploading incorrect files though. I guess I just missed some things on my end as well. Thanks for looking into it. I'll be closing this pull request now.